### PR TITLE
[WIP] Big query based back-end re-write

### PR DESCRIPTION
**Pull request by mayhem**
With ilikecomputer's increased interest in LB and the increasing number of PRs, this back-end re-write ought to get merged ASAP. However, there are a number of things that are still pending and requiring clean-up. In an effort to get this PR merged very soon I would like to:

- Enter bug reports for big things that are still outstanding
- Fix smaller issues immediately that we think need fixing before a merge.

To this effect, I've already entered the following tickets:

1. https://tickets.metabrainz.org/browse/LB-116
2. https://tickets.metabrainz.org/browse/LB-117

This PR:

1. Removes Kafka (\ø/) as a way of communicating incoming listens. Instead Redis is used to move listens from the web server to the components which will consume and store the data.
2. Adds support for using influxdb (a super fast time series database) as the master archival database for listens. influxdb also serves the purpose of deduplicating the incoming data stream, since this is not easy to do with BigQuery.
3. Adds support for the deduplicated data stream from influxdb and writing it to BigQuery. This component is optional and BigQuery credentials are not found, this component will not actively run.
4. Does a lot of other cleanup and small improvements aimed and improving the codebase. There are still a lot of things to do on this front.
5. Is not perfect, but it improves a lot of things. The code is still a ways away from being ready for deployment, but with the help of ilikecomputers and armacolite we can get things moving to a pre-beta release soon. Given this, please expect that this code improves things, but is not perfect. Feel free to add further Jira tickets as todo items for this PR for post merge work.
6. Adds the develop.sh function and improves vagrant support for running LB in an VM. The develop.sh script is handy for starting/restarting containers, but the vagrant support needs improvement/removal in favor of a more pure docker solution in the future.

I'll have to look at the difficulty of merging this PR -- we may need to go with a force push, but recent PRs may be affected by this.

### CODE REVIEW

File: admin/sql/bigquery_schema_listen.sql [hunk 1]
```diff
@@ -0,0 +1,62 @@
+[
```
Comment (alastair) [inline admin/sql/bigquery_schema_listen.sql]: It looks like a schema definition for bigquery. SHould be .json

Comment (mayhem) [inline admin/sql/bigquery_schema_listen.sql]: Yep, fixed in upcoming commit.

File: admin/influx/create_test_db.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import os
+sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), "..", ".."))
+from influxdb import InfluxDBClient
+import config
+
+try:
+    i = InfluxDBClient(host=config.INFLUX_HOST, port=config.INFLUX_PORT, database=config.INFLUX_TEST_DB)
+    i.create_database(config.INFLUX_TEST_DB)
+except Exception as e:
+    print("Creating influx DB failed: ", e)
```
Comment (mayhem) [inline admin/influx/create_test_db.py]: This is what I was alluding to in the PR description -- this is discombobulated and with the volume-less containers, the -test portions of the setup script will simply go away once LB-116 is addressed. But, I would rather not clean any of that up right this second.

File: admin/influx/create_db.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import os
+sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), "..", ".."))
+from influxdb import InfluxDBClient
+import config
+
+try:
+    i = InfluxDBClient(host=config.INFLUX_HOST, port=config.INFLUX_PORT, database=config.INFLUX_DB)
+    i.create_database(config.INFLUX_DB)
+except Exception as e:
+    print("Creating influx DB failed: ", e)
```
Comment (mayhem) [inline admin/influx/create_db.py]: K, makes sense. Will fix.

File: admin/influx/create_db.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+import sys
+import os
+sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), "..", ".."))
+from influxdb import InfluxDBClient
+import config
+
+try:
+    i = InfluxDBClient(host=config.INFLUX_HOST, port=config.INFLUX_PORT, database=config.INFLUX_DB)
+    i.create_database(config.INFLUX_DB)
```
Comment (mayhem) [inline admin/influx/create_db.py]: Fixed. INFLUX_TEST_DB_NAME is not fixed, since that will go away after merge.

**Comment by gentlecat** [comment]
I don't see much point reviewing this since it will take a significant amount of time how all this exactly functions and fits together. I'd say we can merge and create tickets related to specific issues afterwards. Can try setting it up locally with Docker in a bit to see if it works though.

This re-write should've been split up into smaller chunks that added updated features while still keeping old ones until they can be replaced painlessly. Easier to review and manage the transition that way. And we would've been able to keep the latest version deployed and working. Currently any significant development is pretty much on halt until that branch is merged, and I don't even know what commit is actually running on https://listenbrainz.org/ right now.

Not blaming anyone, just pointing out this situation so that we can try to avoid it in the future.

File: admin/influx/create_test_db.py (hunk 2 above)
Comment (gentlecat) [inline admin/influx/create_test_db.py]: OK. 👍

**Comment by mayhem** [comment]
Well, I am clearly to blame, no question about this.

The reason for this mess was when I was in my really bad headspace, I worked on this massive PR in an effort to do something that gave me joy in the MB world. I was working really hard to not quit and say "FUCK EVERYTHING". So, all I did was code and... it worked. Phew.

I've been doing cursory tests in docker to ensure that it works and it does...

**Pull request merged**